proof.go

```diff
@@ -465,6 +465,203 @@ func (proof Proof) VerifyInclusion(h hash.Hash, nid namespace.ID, leavesWithoutN
 	return res
 }
 
+// VerifySubtreeRootInclusion verifies that a set of subtree roots is included in
+// an NMT.
+// Note: This method is Celestia specific.
+// It makes the following assumptions:
+// - The subtree roots are created according to the ADR-013
+// https://github.com/celestiaorg/celestia-app/blob/main/docs/architecture/adr-013-non-interactive-default-rules-for-zero-padding.md
+// - The tree's number of leaves is a power of two
+// Using this method without making sure the above assumptions are respected
+// can return invalid results.
+// The subtreeWidth is also defined in ADR-013.
+// More information on the algorithm used can be found in the ToLeafRanges() method docs.
+func (proof Proof) VerifySubtreeRootInclusion(nth *NmtHasher, subtreeRoots [][]byte, subtreeWidth int, root []byte) (bool, error) {
+	// check that the proof range is valid
+	if proof.Start() < 0 || proof.Start() >= proof.End() {
+		return false, fmt.Errorf("proof range [proof.start=%d, proof.end=%d) is not valid: %w", proof.Start(), proof.End(), ErrInvalidRange)
+	}
```

Comment:
Aren't some of these common to a lot of the verify methods on proof (i.e. verify inclusion or verify namespace) and could be factored out

Reply:
This is similar to: https://github.com/celestiaorg/nmt/pull/260#discussion_r1627799443 for which we created this issue: https://github.com/celestiaorg/nmt/issues/261.

We can do some refactoring, but we need to benchmark them first and see how that impact readability. That's why I preferred to leave all refactoring to be done in a separate PR.

This PR follows exactly the same pattern used in the other proving methods.